packages/caip/src/caip2/caip2.ts

```diff
@@ -32,72 +32,84 @@ type ToCAIP2Args = {
-        [NetworkTypes.OSMOSIS_TESTNET]: ChainReference.OsmosisTestnet
-      }
+const shapeShiftToCAIP2 = Object.freeze({
+  [ChainTypes.Ethereum]: {
+    namespace: ChainNamespace.Ethereum,
+    reference: {
+      [NetworkTypes.MAINNET]: ChainReference.EthereumMainnet,
+      [NetworkTypes.ETH_ROPSTEN]: ChainReference.EthereumRopsten,
+      [NetworkTypes.ETH_RINKEBY]: ChainReference.EthereumRinkeby
+    }
+  },
+  [ChainTypes.Bitcoin]: {
+    namespace: ChainNamespace.Bitcoin,
+    reference: {
+      [NetworkTypes.MAINNET]: ChainReference.BitcoinMainnet,
+      [NetworkTypes.TESTNET]: ChainReference.BitcoinTestnet
+    }
+  },
+  [ChainTypes.Cosmos]: {
+    namespace: ChainNamespace.Cosmos,
+    reference: {
+      [NetworkTypes.COSMOSHUB_MAINNET]: ChainReference.CosmosHubMainnet,
+      [NetworkTypes.COSMOSHUB_VEGA]: ChainReference.CosmosHubVega
     }
-  } as const
+  },
+  [ChainTypes.Osmosis]: {
+    namespace: ChainNamespace.Cosmos,
+    reference: {
+      [NetworkTypes.OSMOSIS_MAINNET]: ChainReference.OsmosisMainnet,
+      [NetworkTypes.OSMOSIS_TESTNET]: ChainReference.OsmosisTestnet
+    }
+  }
+})
 
+export const toCAIP2: ToCAIP2 = ({ chain, network }): string => {
   const namespace: ChainNamespace = shapeShiftToCAIP2[chain].namespace
 
   switch (chain) {
-    case ChainTypes.Cosmos: {
+    case ChainTypes.Ethereum: {
       const referenceMap = shapeShiftToCAIP2[chain].reference
       switch (network) {
-        case NetworkTypes.COSMOSHUB_MAINNET:
-        case NetworkTypes.COSMOSHUB_VEGA:
-        case NetworkTypes.OSMOSIS_MAINNET:
-        case NetworkTypes.OSMOSIS_TESTNET: {
+        case NetworkTypes.MAINNET:
+        case NetworkTypes.ETH_ROPSTEN:
+        case NetworkTypes.ETH_RINKEBY: {
           const reference: ChainReference = referenceMap[network]
           const caip2 = `${namespace}:${reference}`
           return caip2
         }
       }
       break
     }
-
-    case ChainTypes.Ethereum: {
+    case ChainTypes.Bitcoin: {
       const referenceMap = shapeShiftToCAIP2[chain].reference
       switch (network) {
         case NetworkTypes.MAINNET:
-        case NetworkTypes.ETH_ROPSTEN:
-        case NetworkTypes.ETH_RINKEBY: {
+        case NetworkTypes.TESTNET: {
           const reference: ChainReference = referenceMap[network]
           const caip2 = `${namespace}:${reference}`
           return caip2
         }
       }
       break
     }
-
-    case ChainTypes.Bitcoin: {
+    case ChainTypes.Cosmos: {
       const referenceMap = shapeShiftToCAIP2[chain].reference
       switch (network) {
-        case NetworkTypes.MAINNET:
-        case NetworkTypes.TESTNET: {
+        case NetworkTypes.COSMOSHUB_MAINNET:
+        case NetworkTypes.COSMOSHUB_VEGA: {
+          const reference: ChainReference = referenceMap[network]
+          const caip2 = `${namespace}:${reference}`
+          return caip2
+        }
+      }
+      break
+    }
+    case ChainTypes.Osmosis: {
```

Comment:
Can't we reuse the same logic for both swith cases?

Reply:
yeah this is valid

Reply:
I get type errors when I try to consolidate the Cosmos and Osmosis switch statement into one due to how it looks up the reference map.